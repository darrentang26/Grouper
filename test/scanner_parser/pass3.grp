type language = C | Java | Ocaml | Rust | Other of string

let language -> string hasObjects = lambda (lang) match (lang) with
|   (C)        -> "nah"
|   (Java)     -> "heck yeah"
|   (Ocaml)    -> "apparently"
|   (Rust)     -> "no idea"
|   (Other("Grouper")) -> "heck no"
|   (Other(s)) -> s + "++"
end

and language js = "JavaScript" in
    print (hasObjects js)