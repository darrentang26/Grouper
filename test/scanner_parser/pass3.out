type language = C | Java | Ocaml | Rust | Other of string
let language -> string hasObjects = (lang) -> match (lang) with
  | (C) -> "nah"
  | (Java) -> "heck yeah"
  | (Ocaml) -> "apparently"
  | (Rust) -> "no idea"
  | (Other("Grouper")) -> "heck no"
  | (Other(s)) -> s + "++" in
let language js = "JavaScript" in
print: hasObjects js
